File: src/main/java/com/github/nsc/de/compiler/parser/node/loops/DoWhileNode.java
package com.github.nsc.de.compiler.parser.node.loops;

import com.github.nsc.de.compiler.parser.node.Node;
import com.github.nsc.de.compiler.parser.node.Tree;
import com.github.nsc.de.compiler.parser.node.ValuedNode;

public class DoWhileNode implements Node {

    private final Tree body;
    private final ValuedNode condition;


    public DoWhileNode(Tree body, ValuedNode condition) {
        this.body = body;
        this.condition = condition;
    }

    public Tree getBody() {
        return body;
    }

    public ValuedNode getCondition() {
        return condition;
    }

    @Override
    public String toString() {
        return "WhileNode{" +
                "body=" + body +
                ", condition=" + condition +
                '}';
    }
}

File: parser/src/commonMain/kotlin/io/github/shakelang/shake/parser/node/loops/DoWhileNode.kt
package io.github.shakelang.shake.parser.node.loops

import io.github.shakelang.shake.parser.node.Node
import io.github.shakelang.shake.parser.node.Tree
import io.github.shakelang.shake.parser.node.ValuedNode
import io.github.shakelang.parseutils.characters.position.PositionMap

class DoWhileNode(map: PositionMap, val body: Tree, val condition: ValuedNode) : Node(map) {

    override fun toJson(): Map<String, *> =
        mapOf("name" to "DoWhileNode", "body" to body.json, "condition" to condition.json)

}